types/video.js/index.d.ts

```diff
@@ -72,4 +74,137 @@ declare namespace videojs {
 		poster(val?: string): string | Player;
 		playbackRate(rate?: number): number;
 	}
+
+	interface Component {
+		options_: any;
+		el_: HTMLElement;
+		player_: any;
+		isReady_: boolean;
+
+    new(options: any, ready: () => void);
```

Comment:
Did you mean `class Component` and `constructor(...)`? This declares an interface for objects `x` s.t. you can call `new x(...)` and also `x.dispose()` etc.